src/transformers/models/llama/modeling_llama.py

```diff
@@ -285,6 +285,8 @@ def repeat_kv(hidden_states: torch.Tensor, n_rep: int) -> torch.Tensor:
 class LlamaAttention(nn.Module):
     """Multi-headed attention from 'Attention Is All You Need' paper"""
 
+    cached_mask = None
```

Comment:
as seen with you offline, this cannot work as-is due to sharing across model instances

Reply:
Yes will either not cache it at the class level but instance level a tril. Or pass it as kwargs. Jax does not seem to care so should not be too bad

Reply:
Why can't we pass the `attention_mask` just into the `cache.update(...)` function?

Reply:
I'll check that as well, but that doesn't help for all type of attention we have which need a pre-processed mask, will work after the pre-processing tho